### ISSUE TITLE
NameError: name 'inf' is not defined

### ISSUE THREAD

**Issue by zhqu1148980644** [smarie/python-makefun#63]
When the source function has a parameter with a default value `float('inf')`
The `@with_signature` decorator raises this error.

A minor example:

```python
def a(a = float("inf")):
    pass
import inspect
from makefun import with_signature
@with_signature(inspect.signature(a))
def test(*args, **kwargs):
    return a(*args, **kwargs)
```

**Comment by smarie** [comment]
Thanks for the feedback @zhqu1148980644  ! This can also be reproduced using `@wraps` :

```python
def foo(a=float("inf")):
    pass

@wraps(foo)
def test(*args, **kwargs):
    return foo(*args, **kwargs)
```

Being able to compile functions dynamically with non-primitive default values is tricky. It is currently being done in [`get_signature_string`](https://github.com/smarie/python-makefun/blob/d90149d39a1f3913e88442dcc403e2338af3d69f/makefun/main.py#L300). If a non-primitive symbol is seen in the default values it is added to the eval dict. Maybe there is something in your example that does not work with this mechanism, I'll have a look.

**Comment by smarie** [comment]
Oh I see, I was assuming that **all** `int`, `str`, `float`, `bool` have a `repr` that is directly understandable by the compiler. However this is not the case for `inf`, `-inf` and `nan`.

I remove `float` from the types that are assumed safe. I take this opportunity to add `bytes` to the ones that are assumed safe.

**Comment by smarie** [comment]
Fixed in 1.11.1, let me know if the fix works for you !

**Comment by zhqu1148980644** [comment]
Thanks, it worked.
